test/integration/parser_config.cc

```diff
@@ -0,0 +1,240 @@
+/*
+ * Copyright (C) 2020 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include <gtest/gtest.h>
+
+#include "sdf/Filesystem.hh"
+#include "sdf/Model.hh"
+#include "sdf/ParserConfig.hh"
+#include "sdf/Root.hh"
+#include "sdf/SDFImpl.hh"
+#include "sdf/World.hh"
+#include "sdf/parser.hh"
+
+#include "test_config.h"
+
+/////////////////////////////////////////////////
+/// Test global config
+TEST(ParserConfig, GlobalConfig)
+{
+  // The directory used in AddURIPath must exist in the filesystem, so we'll use
+  // PROJECT_SOURCE_PATH
+  const std::string testDir = PROJECT_SOURCE_PATH;
+
+  sdf::addURIPath("file://", testDir);
+  sdf::setFindCallback(
+      [](const std::string &)
+      {
+        return "test/dir2";
+      });
+
+  const auto &config = sdf::ParserConfig::GlobalConfig();
+  auto it = config.URIPathMap().find("file://");
+  ASSERT_NE(config.URIPathMap().end(), it);
+  ASSERT_EQ(1u, it->second.size());
+  EXPECT_EQ(it->second.front(), testDir);
+
+  ASSERT_TRUE(sdf::ParserConfig::GlobalConfig().FindFileCallback());
+  EXPECT_EQ("test/dir2",
+      sdf::ParserConfig::GlobalConfig().FindFileCallback()("empty"));
```

Comment:
how do you feel about adding some expectations about `sdf::findFile`?

~~~
diff --git a/test/integration/parser_config.cc b/test/integration/parser_config.cc
index d6bd2fb3..3f9780ac 100644
--- a/test/integration/parser_config.cc
+++ b/test/integration/parser_config.cc
@@ -51,6 +51,9 @@ TEST(ParserConfig, GlobalConfig)
   ASSERT_TRUE(sdf::ParserConfig::GlobalConfig().FindFileCallback());
   EXPECT_EQ("test/dir2",
       sdf::ParserConfig::GlobalConfig().FindFileCallback()("empty"));
+  // sdf::findFile requires explicitly enabling callbacks
+  EXPECT_EQ("test/dir2", sdf::findFile("empty", false, true));
+  EXPECT_EQ("test/dir2", sdf::findFile("empty", true, true));
 }
 
 /////////////////////////////////////////////////
~~~